[GTK][WPE] Use enable-html5-database runtime flag to control IndexedDB API

pgorszkowski-igalia opened the pull request:
#### c58657f2f72f00a4ad5e05e02449cc9bc3cea6e8
<pre>
[GTK][WPE] Use enable-html5-database runtime flag to control IndexedDB API
<a href="https://bugs.webkit.org/show_bug.cgi?id=261669">https://bugs.webkit.org/show_bug.cgi?id=261669</a>

Reviewed by Michael Catanzaro.

By default IndexedDB API is enabled. To change that we can use runtime flag:
&quot;enable-html5-database&quot; or by calling &quot;webkit_settings_set_enable_html5-database&quot;.

* Source/WebKit/UIProcess/API/glib/WebKitSettings.cpp:
(webkit_settings_get_enable_html5_database):
(webkit_settings_set_enable_html5_database):
* Tools/TestWebKitAPI/Tests/WebKitGLib/TestWebKitWebView.cpp:
(testWebViewEnableHTML5Database):
(beforeAll):

Canonical link: <a href="https://commits.webkit.org/268142@main">https://commits.webkit.org/268142@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/2b2f3625237c74fc1448a730ecf24e8ad5662422

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/18793 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/19134 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/19737 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/20659 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/17593 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/18986 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/22439 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/19272 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/19401 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/19018 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/19156 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/16368 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/21541 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/16378 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/17128 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/23570 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/17405 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/17300 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/21475 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/17899 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/15182 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/16962 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/4472 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/21329 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/17737 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

mcatanzaro reviewed (changes requested):
The existing enable-html5-database setting already exists for enabling/disabled IndexedDB. If that no longer works after [254772@main](https://commits.webkit.org/254772@main), then it's a serious regression that we need to fix before making any further changes.

 * This is missing `Since: 2.44` tags on the new methods and the new property
 * I would rename `enable-indexeddb-api` to just `enable-indexeddb` to simplify the name.
 * You should definitely deprecate the existing html5-database setting if adding this replacement.
 * I'm not convinced it really makes sense to add a new setting that has effectively no behavior difference vs. the existing setting. WebSQL is no longer supported, and the html5-database setting is documented to only affect IndexedDB nowadays. Your new setting has a better name, but that's not really sufficient justification for new APIs.

I think my preference would be:

 * Fix, make sure to unbreak the existing html5-database setting if it's broken
 * Rename the setting in a future API version, e.g. by adding a new preprocessor guard `ENABLE_FUTURE_GLIB_API` and making the changes conditional on that. Then 5-10 years from now, if WebKitGTK still exists, we can flip that flag and get the nicer names. There are a few other APIs that I'd like to rename....

mcatanzaro reviewed (commented):
OK, the changes look good now. The commit message needs to be updated since you've completely rewritten this patch, though.

Does the test fail without your changes to WebKitSettings.cpp? My assumption is that the databasesEnabled setting disables all other database settings, so that it would still effectively enable/disable IndexedDB? If so, then your test should pass either way and the changes to WebKitSettings.cpp are optinal. If not, then the enable-html5-database setting has been broken since 2.40, which is unfortunate. Oh well, at least we'll have a good test now.

pgorszkowski-igalia committed:
[GTK][WPE] Use enable-html5-database runtime flag to control IndexedDB API
https://bugs.webkit.org/show_bug.cgi?id=261669

Reviewed by Michael Catanzaro.

By default IndexedDB API is enabled. To change that we can use runtime flag:
"enable-html5-database" or by calling "webkit_settings_set_enable_html5-database".

* Source/WebKit/UIProcess/API/glib/WebKitSettings.cpp:
(webkit_settings_get_enable_html5_database):
(webkit_settings_set_enable_html5_database):
* Tools/TestWebKitAPI/Tests/WebKitGLib/TestWebKitWebView.cpp:
(testWebViewEnableHTML5Database):
(beforeAll):

Canonical link: https://commits.webkit.org/268142@main
